src/interface/idl.rs

```diff
@@ -0,0 +1,594 @@
+    Enum(EnumDefinition<'src>),
+    Record(RecordDefinition<'src>),
+    Class(ClassDefinition<'src>),
+    Function(FunctionDefinition<'src>),
+}
+
+impl<'src> ModuleMember<'src> {
+    fn parse_from(input: &'src str) -> ParseResult<'src, Self> {
+        alt((
+            map(EnumDefinition::parse_from, |defn| Self::Enum(defn)),
+            map(RecordDefinition::parse_from, |defn| Self::Record(defn)),
+            map(ClassDefinition::parse_from, |defn| Self::Class(defn)),
+            map(FunctionDefinition::parse_from, |defn| Self::Function(defn)),
+        ))(input)
+    }
+}
+
+/// The [`enum`](EnumDefinition) keyword defines a C-style enum as a new named type.
+/// Each enum has a fixed number of options which can be specified by name. It is not
+/// possible to set or to example the internal representation of the enum.
+///
+/// ```text
+/// enum exampleNameOfEnum {
+///      /* An identifier for each option, separated by commas */
+///     OPTION_ONE,
+///     OPTION_TWO,
+/// }
+/// ```
+pub(crate) struct EnumDefinition<'src> {
+    name: &'src str,
+    members: Vec<&'src str>,
+}
+
+impl<'src> EnumDefinition<'src> {
+    fn parse_from(input: &'src str) -> ParseResult<'src, Self> {
+        map(
+            tuple((
+                kwd("enum"),
+                identifier,
+                sym("{"),
+                comma_separated_list(identifier),
+                sym("}"),
+                sym(";"),
+            )),
+            |(_, name, _, members, _, _)| EnumDefinition { name, members },
+        )(input)
+    }
+}
+
+pub(crate) struct RecordDefinition<'src> {
+    name: &'src str,
+    fields: Vec<FieldDefinition<'src>>,
+}
+
+impl<'src> RecordDefinition<'src> {
+    fn parse_from(input: &'src str) -> ParseResult<'src, Self> {
+        map(
+            tuple((
+                kwd("record"),
+                identifier,
+                sym("{"),
+                many0(FieldDefinition::parse_from),
+                sym("}"),
+                sym(";"),
+            )),
+            |(_, name, _, fields, _, _)| RecordDefinition { name, fields },
+        )(input)
+    }
+}
+
+pub(crate) struct FieldDefinition<'src> {
+    name: &'src str,
+    type_: TypeReference<'src>,
+}
+
+impl<'src> FieldDefinition<'src> {
+    fn parse_from(input: &'src str) -> ParseResult<'src, Self> {
+        map(
+            tuple((TypeReference::parse_from, identifier, sym(";"))),
+            |(type_, name, _)| FieldDefinition { name, type_ },
+        )(input)
+    }
+}
+
+pub(crate) enum TypeReference<'src> {
+    Boolean,
+    U8,
+    I8,
+    U16,
+    I16,
+    U32,
+    I32,
+    U64,
+    I64,
+    Float,
+    Double,
+    Nullable(Box<TypeReference<'src>>),
+    Sequence(Box<TypeReference<'src>>),
+    Named(&'src str),
+}
+
+impl<'src> TypeReference<'src> {
+    fn parse_from(input: &'src str) -> ParseResult<'src, Self> {
+        let (input, type_) = alt((
+            map(kwd("boolean"), |_| TypeReference::Boolean),
```

Comment:
Another gratuitous suggestion: how do you feel about `bool`, `f32`, and `f64` to match Rust?

(And thanks for not calling `u64` `unsigned long long`! 😆❤️)